controllers/gateway/resource.go

```diff
@@ -88,7 +88,7 @@ func (ctx *gatewayContext) buildLegacyServiceResource() (*corev1.Service, error)
 func (ctx *gatewayContext) makeDeploymentSpec() (*appv1.DeploymentSpec, error) {
 	// Deprecated spec, will be unsupported soon.
 	if ctx.gateway.Spec.Template.Spec != nil {
-		ctx.logger.WithField("name", ctx.gateway.Name).WithField("namespace", ctx.gateway.Namespace).Warn("spec.template.spec is DEPRECATED, it will be unsupported soon, please use spec.template.container")
+		ctx.logger.WithField("name", ctx.gateway.Name).WithField("namespace", ctx.gateway.Namespace).Warn("spec.template.spec is DEPRECATED, it will be unsupported soon, please use spec.template.containers")
```

Comment:
Why change to `containers`? it should be `container`.

Reply:
`container` is what generates the log floods, ` containers` has no problem

Reply:
I didn't get you.

1. Making a log content change will not change anything.
2. The reason why the warning appears is because a to be deprecated spec `spec.template.spec.xxx` is being used, it recommends to use new spec `spec.template.container`.
3. `spec.template.containers` is never supported, I guess you meant `spec.template.spec.containers`, which applies to the one to be deprecated.